### PR TITLE
TRUNK-6238: Drug Ingredients - Switching from Hibernate Mapping to JPA annotations

### PR DESCRIPTION
**Pull request by wikumChamith**
## Description of what I changed

Migrating from Hibernate mapping to JPA annotations for the Drug Ingredients table.

## Issue I worked on

see https://openmrs.atlassian.net/browse/TRUNK-6238

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can we include the ticket id in the commit message?

**Comment by wikumChamith** [comment]
@dkayiwa I updated the commit message.

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 1]
```diff
@@ -18,49 +18,8 @@
...
-import org.openmrs.DrugOrder;
-import org.openmrs.Encounter;
-import org.openmrs.FreeTextDosingInstructions;
-import org.openmrs.GlobalProperty;
-import org.openmrs.Location;
-import org.openmrs.LocationAttributeType;
-import org.openmrs.MedicationDispense;
-import org.openmrs.Obs;
-import org.openmrs.Order;
+import org.openmrs.*;
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:21]: https://openmrs.atlassian.net/wiki/spaces/docs/pages/25477044/Java+Conventions#JavaConventions-Donotusewildcardimports

File: api/src/main/java/org/openmrs/DrugIngredient.java [hunk 2]
```diff
@@ -9,21 +9,39 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
 /**
  * DrugIngredient
  */
-public class DrugIngredient extends BaseOpenmrsObject implements java.io.Serializable, OpenmrsObject {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/DrugIngredient.java:15]: Is there any reason why you dropped only `java.io.Serializable` but not `OpenmrsObject`?

**Comment by wikumChamith** [comment]
@dkayiwa I updated this PR.

**Comment by dkayiwa** [comment]
@wikumChamith can you look into the merge conflicts?

**Comment by wikumChamith** [comment]
@dkayiwa I fixed them.

**Comment by wikumChamith** [comment]
@dkayiwa I am not getting the test failures locally.

**Comment by wikumChamith** [comment]
> @dkayiwa I am not getting the test failures locally.

@dkayiwa It got fixed after pushing again :sweat_smile:

**Pull request merged**